### PR TITLE
Add test suite for tasks_controller.

### PR DESCRIPTION
**Pull request by ishmam-iqbal-sazim**
### Purpose
- Add unit tests to ensure code reliability.

### Changes
- Add unit tests for each endpoints.
- Refactor DestroyAllTasks flow and add set_task_type interactor.
- Refactor tasks controller and interactors handling the tasks to properly handle errors.

### CODE REVIEW

File: app/interactors/tasks/set_task_type.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Tasks::SetTaskType
+    include Interactor
+
+    def call
+        if context.type == "completed"
```
Comment (ragib-hossain-sazim) [inline app/interactors/tasks/set_task_type.rb]: why are you assigning params in the interactor?? on why is this change required?

Comment (Md-Ishmam-Iqbal) [reply app/interactors/tasks/set_task_type.rb]: The destroy_all endpoints works for two purposes: to destroy all completed tasks or to destroy all archived tasks. When the endpoint is called the type (completed or archived) params is being sent. Based on the type, the task_params should be set to either (is_completed: true) or (is_deleted: true). To handle this logic outside the controller, I included this interactor.

Comment (Md-Ishmam-Iqbal) [reply app/interactors/tasks/set_task_type.rb:5]: The destroy_all endpoints works for two purposes: to destroy all completed tasks or to destroy all archived tasks. When the endpoint is called the type (completed or archived) params is being sent. Based on the type, the task_params should be set to either (is_completed: true) or (is_deleted: true). To handle this logic outside the controller, I included this interactor.

Comment (ragib-hossain-sazim) [reply app/interactors/tasks/set_task_type.rb]: why can't we pass this from frontend as params? whatever params is passed the destroy_all just handles that. If it's completed tasks then completed otherwise archived.

File: app/interactors/tasks/destroy_all_tasks.rb [hunk 2]
```diff
@@ -2,12 +2,7 @@ class Tasks::DestroyAllTasks
...
     def call
-        if context.type == "completed"
-            completed_tasks = context.tasks
-            context.fail!(errors: "Something went wrong. Completed tasks delete failed.") unless Task.destroy(completed_tasks.ids)
-        elsif context.type == "archived"
-            archived_tasks = context.tasks
-            context.fail!(errors: "Something went wrong. Archived tasks delete failed.") unless Task.destroy(archived_tasks.ids)
-        end
+        tasks_to_be_deleted = context.tasks
+        context.fail!(errors: "Tasks delete failed") unless Task.destroy(tasks_to_be_deleted.ids)
```
Comment (ragib-hossain-sazim) [inline app/interactors/tasks/destroy_all_tasks.rb]: Delete action should be performed in a separate line.

File: test/controllers/api/v1/tasks_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        "due_date" => task.due_date,
+        "is_completed" => task.is_completed,
+        "is_deleted" => task.is_deleted
+      }
+    end
+
+    response_body = JSON.parse(response.body)
+
+    assert_response :ok
+    assert_equal expected_response, response_body
```
Comment (ragib-hossain-sazim) [inline test/controllers/api/v1/tasks_controller_test.rb]: match length of response

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending comments
